### ISSUE TITLE
[API] Order not found after addressing

### ISSUE THREAD

**Issue by vvasiloi** [Sylius/Sylius#13106]
**Sylius version affected**: 1.9 and above

**Description**  
After the order is addressed, it becomes unusable.
Some routes will return 404 Not Found, other 500 Internal Server Error.

**Steps to reproduce**  

1. Pick up a new cart

```shell
curl -X 'POST' \
  'https://demo.sylius.com/api/v2/shop/orders' \
  -H 'accept: application/ld+json' \
  -H 'Content-Type: application/ld+json' \
  -d '{
  "localeCode": "en_US"
}'
```

2. Add a product to the cart

```shell
curl -X 'PATCH' \
  'https://demo.sylius.com/api/v2/shop/orders/{tokenValue}/items' \
  -H 'accept: application/ld+json' \
  -H 'Content-Type: application/merge-patch+json' \
  -d '{
  "productVariant": "/api/v2/shop/product-variants/{code}",
  "quantity": 1
}'
```

3. Address

```shell
curl -X 'PATCH' \
  'https://demo.sylius.com/api/v2/shop/orders/{tokenValue}/address' \
  -H 'accept: application/ld+json' \
  -H 'Content-Type: application/merge-patch+json' \
  -d '{
  "email": "shop@example.com",
  "billingAddress": {
    "countryCode": "US",
    "firstName": "John",
    "lastName": "Doe",
    "street": "Test",
    "city": "Test",
    "postcode": "1234"
  },
  "shippingAddress": {
    "countryCode": "US",
    "firstName": "John",
    "lastName": "Doe",
    "street": "Test",
    "city": "Test",
    "postcode": "1234"
  }
}
'
```

4. Get the order

```shell
curl -X 'GET' \
  'https://demo.sylius.com/api/v2/shop/orders/{tokenValue}' \
  -H 'accept: application/ld+json'
```

Any other order endpoint can be used at point 4, the result will be the same: 404 Not Found.
You can try addressing it again, for example.

**Possible Solution**  

WIP

---

UPDATE:

Looks like it only happens when the submitted email belongs to a registered user, like `shop@example.com` from my test on the demo shop.
Providing a JWT does not fix the issue.

**Comment by Rafikooo** [comment]
Hello @vvasiloi,
Thanks for creating the issue. I have reproduced this case on the stable demo which is Sylius 1.11 by the time of writing and confirm that the issue exists. However, on the master demo (Sylius 1.12) everything's fine. As API is in an experimental phase and maintenance of Sylius 1.11 is coming to an end, I will close this issue 📪

**Comment by vvasiloi** [comment]
@Rafikooo do you know if there's a test for this, to prevent regretion?

**Comment by Rafikooo** [comment]
It fixes the case that you described in Sylius 1.11. Then we should upmerge only the contract test as the code has been refactored from Sylius 1.12 upwards
https://github.com/Sylius/Sylius/pull/14596
